lstchain/mc/sensitivity.py

```diff
@@ -68,35 +70,42 @@ def process_mc(simtelfile, dl2_file):
     sim_par = read_sim_par(source)
     events = pd.read_hdf(dl2_file)
 
-    e_reco = 10**events.mc_energy * u.GeV
-    # n_reco = e_reco.shape[0]
+    e_reco = 10**events.mc_energy.to_numpy() * u.GeV
+
     gammaness = events.gammaness
-    theta2 = (events.src_x - events.src_x_rec)**2 + \
-        (events.src_y - events.src_y_rec)**2 * u.deg**2
+
+    if events.iloc[0].mc_type==0:
```

Comment:
I don't quite understand this condition.
- only the type of the first event is checked - shouldn't one filter all the events to keep, .e.g `events[events.mc_type==0]` ?
- if not a gamma, why theta2 is computed relative to the camera center?

Reply:
- Only the first event is checked because as it is coded now the files that you give should contain only events from one type of particle (a file of gammas or a file of protons), but yes I guess they can be filtered  in case at some point we give mixed files. 

- At first I was using the same theta2 definition, but @moralejo and @rlopezcoto explained that we need to calculate the rate of protons in a region close to the poisition of the gamma source.

Reply:
- It's not more complicated to filter and will avoid future issues. If you could change that please.
- still don't get it, if you want the rate of protons close to the position of the gamma source, you should also compute theta2 relative to `events.src_x[events.mc_type==0]`, no ?

Reply:
I'm taking a look at the filtering thing, but I don't see it clear, because the function is constructed in a way it's assumed that you are going to pass to it only events of one type (since it takes a simtel file) and if we apply a filter, it means that you could pass events of two types and then return two sets of theta2 values, and one of them won't be related to the simtelfile... I think is a bit messy. Anyway I think this function will be changed once we have the new DL1b format in which the mc info is stored in the .h5 file so it won't be necessary to read the simtelfile.

Reply:
Yes, I think the confusion comes from the fact of using a single function to make two very different processing.
The theta2 calculated here for gammas is of a different nature than the one calculated for hadrons (and electrons?) and I got confused by it.
Maybe we should consider breaking it down to different functions.

Reply:
There is a `ring_theta2` and a `source_theta2`.
Then they are applied to different particle types.

Reply:
I had a chat with @vuillaut about something else and we also talked about this... probably the best would be to propagate the information of `events.src_x` and `events.src_x_rec` (also the y coordinate) further, so we can compute later the theta2 for gammas and the angular distance for protons. We can program a ring for the proton (electron) extraction later, is that ok?

Reply:
Related to this... `events.src_x` has units of `m`, right? If it is like this, then the theta2 is completely wrong...